### PR TITLE
Exception packet

### PR DESCRIPTION
**Pull request by rbaryanai**

### CODE REVIEW

File: lib/netdev-rte-offloads.c [hunk 1]
```diff
@@ -411,6 +414,10 @@ static void netdev_rte_port_free(struct netdev_rte_port * rte_port)
                         CONST_CAST(struct cmap_node *, &rte_port->node), hash);
             netdev_rte_free_table_id(rte_port->table_id);
             netdev_rte_reserved_mark_free(rte_port->special_mark);
+            cmap_remove(&mark_to_rte_port, CONST_CAST(struct cmap_node *,
+                        &rte_port->mark_node),
```
Comment (asafpenso) [inline lib/netdev-rte-offloads.c:419]: For better readability can you do the following:
cmap_remove(&mark_to_rte_port,
                       CONST_CAST(struct cmap_node *, &rte_port->mark_node),

File: lib/netdev-rte-offloads.c [hunk 2]
```diff
@@ -1471,3 +1483,51 @@ netdev_dpdk_flow_del(struct netdev *netdev OVS_UNUSED, const ovs_u128 *ufid,
...
+                    rte_port->netdev->netdev_class->pop_header(packet);
+                    packet->md.in_port.odp_port = rte_port->port_no;
+                    reset_dp_packet_checksum_ol_flags(packet);
+                }
+            }
+            break;
+        case RTE_PORT_TYPE_NONE:
+        case RTE_PORT_TYPE_DPDK:
+            break;
+
```
Comment (asafpenso) [inline lib/netdev-rte-offloads.c:1508]: isn't default always required by checkpatch?

File: lib/netdev-rte-offloads.c [hunk 3]
```diff
@@ -1471,3 +1483,51 @@ netdev_dpdk_flow_del(struct netdev *netdev OVS_UNUSED, const ovs_u128 *ufid,
...
+                if (rte_port->netdev->netdev_class->pop_header) { 
+                    rte_port->netdev->netdev_class->pop_header(packet);
+                    packet->md.in_port.odp_port = rte_port->port_no;
+                    reset_dp_packet_checksum_ol_flags(packet);
+                }
+            }
+            break;
+        case RTE_PORT_TYPE_NONE:
+        case RTE_PORT_TYPE_DPDK:
+            break;
```
Comment (asafpenso) [inline lib/netdev-rte-offloads.c]: How about a print? info/debug/warn?

File: lib/netdev-rte-offloads.c [hunk 4]
```diff
@@ -1471,3 +1483,51 @@ netdev_dpdk_flow_del(struct netdev *netdev OVS_UNUSED, const ovs_u128 *ufid,
...
+                // no the port-id so we just pop it here.
+                if (rte_port->netdev->netdev_class->pop_header) { 
+                    rte_port->netdev->netdev_class->pop_header(packet);
+                    packet->md.in_port.odp_port = rte_port->port_no;
+                    reset_dp_packet_checksum_ol_flags(packet);
+                }
+            }
+            break;
+        case RTE_PORT_TYPE_NONE:
+        case RTE_PORT_TYPE_DPDK:
```
Comment (asafpenso) [inline lib/netdev-rte-offloads.c:1505]: How about a print? info/debug/warn?

File: lib/netdev-rte-offloads.c [hunk 5]
```diff
@@ -1471,3 +1483,51 @@ netdev_dpdk_flow_del(struct netdev *netdev OVS_UNUSED, const ovs_u128 *ufid,
...
+
+
+static void netdev_rte_port_preprocess(struct netdev_rte_port * rte_port,
+                                       struct dp_packet *packet)
+{
+    switch (rte_port->rte_port_type) {
+        case RTE_PORT_TYPE_VXLAN:
+            {
+                // VXLAN table failed to match on HW. we do however
+                // no the port-id so we just pop it here.
```
Comment (asafpenso) [inline lib/netdev-rte-offloads.c]: typo "no" -> "know"

File: lib/netdev-rte-offloads.c [hunk 6]
```diff
@@ -1471,3 +1483,51 @@ netdev_dpdk_flow_del(struct netdev *netdev OVS_UNUSED, const ovs_u128 *ufid,
...
+        case RTE_PORT_TYPE_DPDK:
+            break;
+
+    }
+    return;
+}
+
+
+/**
+ * @brief - we got a packet with special mark, means we need to run
```
Comment (asafpenso) [inline lib/netdev-rte-offloads.c:1515]: dont you want to mention that special mark is received when we get a miss on a table and de-facto this is an exception handling?

File: lib/netdev-rte-offloads.c [hunk 7]
```diff
@@ -86,6 +86,9 @@ struct netdev_rte_port {
   struct cmap_node all_node;  // map by netdev
   struct netdev * netdev;
 
+  struct cmap_node mark_node;
+  uint32_t mark;
```
Comment (asafpenso) [inline lib/netdev-rte-offloads.c:90]: I dont see a use of this field

**Review by asafpenso — COMMENTED** [review]
Please see my comments.

File: lib/netdev-rte-offloads.c (hunk 2 above)
Comment (rbaryanai) [reply lib/netdev-rte-offloads.c:1508]: on enum you get warning from compiler if you don't check all possible values so therefor there is no need for such test,

File: lib/netdev-rte-offloads.c (hunk 4 above)
Comment (rbaryanai) [reply lib/netdev-rte-offloads.c:1505]: added warn, because don't expect to reach here.

File: lib/netdev-rte-offloads.c (hunk 7 above)
Comment (rbaryanai) [reply lib/netdev-rte-offloads.c:90]: it used for saving and restoring from the hash by mark

**Pull request closed without merging**